src/server/server_family.cc

```diff
@@ -1067,6 +1081,50 @@ void ServerFamily::Client(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendError(UnknownSubCmd(sub_cmd, "CLIENT"), kSyntaxErrType);
 }
 
+void ServerFamily::Cluster(CmdArgList args, ConnectionContext* cntx) {
+  ToUpper(&args[1]);
+  string_view sub_cmd = ArgS(args, 1);
+
+  if (is_emulated_cluster_ == false) {
+    return (*cntx)->SendError("CLUSTER commands demands the `cluster_mode` flag set to `emulated`");
+  }
+
+  if (sub_cmd == "HELP") {
+    string_view help_arr[] = {
+        "CLUSTER <subcommand> [<arg> [value] [opt] ...]. Subcommands are:",
+        "SLOTS",
+        "    Returns details about which cluster slots map to which instances.",
+        "HELP",
+        "    Prints this help.",
+    };
+    return (*cntx)->SendSimpleStrArr(help_arr, ABSL_ARRAYSIZE(help_arr));
+  }
+
+  if (sub_cmd == "SLOTS") {
+    (*cntx)->StartArray(1);    // Replying just one cluster
+    (*cntx)->StartArray(3);    // Should return a replica with master info?
+    (*cntx)->SendLong(0);      // start slot
+    (*cntx)->SendLong(16383);  // end slot
+    (*cntx)->StartArray(3);    // master slot network information
+    string cluster_announce_ip = GetFlag(FLAGS_cluster_announce_ip);
+    if (cluster_announce_ip.empty()) {
+      // On Redis documentation is specified to return
+      // either an IP address, hostname, or NULL
+      // But redis-py library tries to len() this an errors if
```

Comment:
an -> and

Reply:
bonus points - to see whether it passes with at least another client library. We have ioredis tests (see tests/README.md) but they will require tuning `tests/integration/.run_ioredis_valid_test.sh` (see https://github.com/luin/ioredis/tree/main/test/cluster)

Reply:
I'm currently getting timeout on the `./run_ioredis_on_docker.sh`.
![image](https://user-images.githubusercontent.com/70452831/203658801-944145e2-706b-45c3-97ab-d20c164c1984.png)

I'll try to run the tests directly from the project.

Reply:
Are you running DF as well? normally this happens when DF is not working or you have some other network issue between the container and the host

Reply:
Yes, I am but I'm also in an unusual environment. I have a Windows machine and work with dragonfly in WSL so my dragonfly instance runs on WSL and the tests run on Docker on Windows. I'll try to solve this on the weekend.

Reply:
@lsvmello I am more than happy to provide you a cloud dev instance. Just send me your public key (here or on discord)

Reply:
This PR will enable running with ioredis. The issue right now is with the info command as the cluster slots command is working